src/ApiCallable.php

```diff
@@ -121,9 +121,12 @@ private static function setCustomHeader($callable, $headerDescriptor)
                 throw new InvalidArgumentException('Metadata argument is not found.');
             } else {
                 $metadata = $params[self::GRPC_CALLABLE_METADATA_INDEX];
-                $params[self::GRPC_CALLABLE_METADATA_INDEX] =
-                    array_merge($metadata, $headerDescriptor->getHeader());
-                call_user_func_array($callable, $params);
+                $headers = $headerDescriptor->getHeader();
+                if (array_key_exists('headers', $metadata)) {
+                    $headers = array_merge($headers, $metadata['headers']);
```

Comment:
Do we want the values from $headers to overwrite the existing values in $metadata['headers']? Because it seems that array_merge takes values from the later argument when there is duplication of keys.
http://php.net/manual/en/function.array-merge.php